### DISCUSSION TITLE
コミットメッセージに絵文字をつけてもいいかも

### DISCUSSION THREAD

**Discussion by kazzyfrog** [first-contributions-ja/first-contributions-ja.github.io#24] [12: 新機能の提案]
現在のコミットメッセージ： `prefix: 内容 #issue番号`

prefixは下記：
- feat: 新機能追加関連
- docs: ドキュメント関連
- fix: バグ修正関連
- その他は適宜追加

prefixに絵文字を対応させたら、見やすいかも：`prefix: :emoji: 内容 #issue番号`
- feat: :sparkles: 新機能
- docs: :books: ドキュメント関連
- fix: :bug: バグ修正関連

**Comment by pss-aileen** [comment]
いいですね！個人的には絵文字を先頭につけると統一感が出て良いのかなと思ったりします！

以下個人的なリポでコミットだいぶ適当なのですが、こんな感じになります。

![スクリーンショット 2024-03-13 8 59 49](https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/8f8b257a-d773-46e6-a29e-aea56253a235)

**Comment by kagomen** [comment]
先頭につけるの良いですね。
ご存知かと思いますが、gitmojiなるものがあるみたいです！
https://gitmoji.dev/

**Reply by pss-aileen** [reply]
これ見たことはあったんですが、これは絵文字のガイド的なものですかね？💭

**Reply by kagomen** [reply]
ですです！
VSCodeだと拡張機能もあるみたいです。
[gitmoji絵文字のテンプレート -Zenn](https://zenn.dev/ukigumo_shiina3/articles/gitmoji-template)

**Comment by kazzyfrog** [comment]
確かに絵文字を先頭につけるのいいですね 😎 
もし問題がなければ、issueに追加できればと思います:
CONTRIBUTINGガイドの、コミットメッセージの内容を絵文字付きに修正する

@kagomen 
せっかくなので、良かったらお手隙の際にissue立ててみてください！

**Reply by kagomen** [reply]
承知しました！
やってみます🌈

**Reply by kagomen** [reply]
こちらすでにイシューがあったので、イシューの内容を編集しました👮

[CONTRIBUTING.mdの修正（複数） #29](https://github.com/first-contributions-ja/first-contributions-ja.github.io/issues/29)
<img width="658" alt="スクリーンショット 2024-03-14 11 58 56" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/0cf4d862-8e2d-4634-b30c-bca2cfbf8d03">

**Reply by pss-aileen** [reply]
あれだったら今決まっている
git commit origin HEADとコミットメッセージの絵文字の件をまとめてissue切り分けても良いかもです！
こちらは変更するものが明確決まっているので！

あ、でもタスクをシンプルにするために
絵文字の件だけ切り出した方がいい気がしました！
なので絵文字の件でのissueを願いします！

**Reply by kagomen** [reply]
わかりました！

**Comment by kagomen** [comment]
@kazzyfrog @pss-aileen @kana1166 
絵文字の使用に関して疑問がわいたので、ご意見いただきたいです！

gitmojiを使う場合、
- 手打ち入力
- VSCodeの拡張機能（こちらで慣れるとターミナル操作を覚えられないかも）
- cliをインストール

の方法があるかと思いますが、これは各自自由にって感じで良いでしょうか❔

またはそもそもgitmojiを使わずに、
- プロジェクト独自の絵文字を使う
- 自由に絵文字を使用する（絵文字の役割は完全に画面の盛り上げとする）

などの選択肢もありますが、どう思われますか❔

**Reply by kagomen** [reply]
cliをインストールして使用する方法は以下の記事を参照
- [gitmojiの使い方 -Zenn](https://zenn.dev/ogakuzuko/articles/a160fdd8b4b3b8)

実際に`npm i`して使ってみました。
<img width="404" alt="スクリーンショット 2024-03-14 13 54 04" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/07b17167-ac1a-421b-a4ce-a1f39147da37">

導入自体は簡単だったのですが、
`git commit -m "コミットメッセージ"`
の入力はせずに、gitmoji側で調整してくれる感じだったので、どうなのかな〜と

**Reply by pss-aileen** [reply]
### gitmojiの入力方法について

自由で良いのかな？と考えたのですが...
cli入れた方が統一しやすそうですよね...その場で適切な絵文字を選べそうなので...！

ただ、コミットの仕方みんな色々だと思うので
結局はどのprefixにどの絵文字を使うかだけ決めておけば良いのかなぁと思います。

どんなんだろう。他の方の意見も聞きたいです。
あとは自分たちの使い勝手次第かなと。

### gitmoji通りにするか

gitemoji通りでも良いと思いますが
プロジェクト独自でも良いとも思っています...笑

単純に絵文字をつかうと楽しい感じになるし、
判別しやすくなるので良いなと思っていて

なので個人としてはわりと絵文字はどれになってもOKです！
最初の kazzyfrog さんがあげてくださった絵文字でも良いかも！と思っています！

**Reply by kagomen** [reply]
ではCONTRIBUTING.mdに記載する内容としては、
- 記述例としていくつか代表的なものを載せる
- 他は自由でOK

の2点で大丈夫そう、ですかね❔
それでよければCONTRIBUTING.mdを変更して明日にでもプルリク出しますが、Aileenさん編集中だったりしますか❔❔（コンフリクトこわい）

**Reply by pss-aileen** [reply]
そうですね、現状

> * feat: ✨ 新機能
> * docs: 📚 ドキュメント関連
> * fix: 🐛 バグ修正関連

があるのと、あとstyle: スタイル関連
もあって良いかな？と思ったのでそちらもあれば！
絵文字おまかせします...笑
gitmojiでは💄ですかね...！

他のprefixは新しく出てきたらまたその時追加でよいのかなと思います！

自分はCONTRIBUTING.mdを編集は現状してないので大丈夫かと思います！！👌

まぁコンフリクト起きてもなんとかなるでしょう...笑！

**Reply by kagomen** [reply]
了解です！

**Comment by pss-aileen** [comment]
`chore` の追加希望です！

choreとは雑事（カテゴライズする必要のないもの）という意味です。
[コミットメッセージ規約](https://gist.github.com/minop1205/5fc4f6ef0ec89fb1738833ba25ae00a0)

ライブラリ追加したりするときに使えるかな...と。

とりあえず独り言のように残しておきます。
labelはちょっと勝手に作成済みです。
